File: Python/Easy/1945.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        totalSum = sum(int(digit) for char in s for digit in str(ord(char) - ord("a") + 1))

        while k > 1:
            temp = 0

            while totalSum > 0:
                temp += totalSum % 10
                totalSum //= 10

            totalSum = temp
            k -= 1

        return totalSum

File: Java/Easy/1945.java
class Solution {
    public int getLucky(String s, int k) {
        int sum = 0;
        int num;
        int len = s.length();

        int temp;

        // get the initial digit sums
        for (int i = 0; i < len; i++) {
            num = s.charAt(i) - 'a' + 1;

            // multi-digit char
            if (num >= 10) {
                sum += num / 10;
            }

            sum += num % 10;

        }

        // iterate through k
        while (k > 1) {
            temp = 0;
            while (sum > 0) {
                temp += sum % 10;
                sum /= 10;
            }

            sum = temp;
            k--;
        }

        return sum;
    }
}